File: src/Dota2Analyzer.Core/Analysis/MatchSelection.cs
using Dota2Analyzer.Core.Models.Analysis;

namespace Dota2Analyzer.Core.Analysis;

internal static class MatchSelection
{
    public static List<MatchAnalysisResult> SelectDesired(List<MatchAnalysisResult> analyses, int desiredCount, bool onlyPos1)
    {
        if (desiredCount <= 0) return [];

        if (onlyPos1)
        {
            return analyses.Where(a => a.IsPosition1).Take(desiredCount).ToList();
        }

        return analyses.Take(desiredCount).ToList();
    }
}

File: dota2-analyzer-core/src/main/java/com/dota2analyzer/core/analysis/MatchSelection.java
package com.dota2analyzer.core.analysis;

import com.dota2analyzer.core.model.analysis.MatchAnalysisResult;

import java.util.ArrayList;
import java.util.List;

public final class MatchSelection {

    private MatchSelection() {}

    public static List<MatchAnalysisResult> selectDesired(List<MatchAnalysisResult> analyses, int desiredCount, boolean onlyPos1) {
        if (desiredCount <= 0) return new ArrayList<>();

        if (onlyPos1) {
            List<MatchAnalysisResult> result = new ArrayList<>();
            for (MatchAnalysisResult a : analyses) {
                if (a.isPosition1()) {
                    result.add(a);
                    if (result.size() >= desiredCount) break;
                }
            }
            return result;
        }

        return new ArrayList<>(analyses.subList(0, Math.min(desiredCount, analyses.size())));
    }
}
